internal/webhooks/cluster.go

```diff
@@ -443,3 +481,40 @@ func machineDeploymentClassOfName(clusterClass *clusterv1.ClusterClass, name str
 	}
 	return nil
 }
+
+// validateCIDRBlocks validates that:
+// 1) No more than two CIDR blocks are specified.
+// 2) If two are specified the blocks must be from different IP families i.e. one IPv4 and one IPv6.
+// 3) The CIDR ranges are valid CIDR ranges.
+func validateCIDRBlocks(cidrs []string) error {
```

Comment:
Thx!

The most straightforward and simple implementation of this validation that I saw up until now :)